sys/vane/ames.hoon

```diff
@@ -627,50 +296,62 @@
               :-  %fast
               %^  cat  7
                 p.u.yed.caq.dur
-              (en:r:cluy q.u.yed.caq.dur (jam ham))
+              (en:cub:cluy q.u.yed.caq.dur (jam ham))
             ?:  &(=(~ lew.wod.dur) =(%back -.ham))
               [wasp ..wisp]
-            =^  tuy  +>.$
-              ?:(=(~ lew.wod.dur) [*code +>.$] (griz now))
-            :_  ..wisp
-            =+  yig=sen
-            ::  =+  bil=`wyll`(pare wyl.dur law.saf)    ::  XX not set
-            =+  bil=law.saf                             ::  XX send whole wyll
-            =+  hom=(jam ham)
+            =/  bil=(unit deed)
+              ?.  ?|  ?=(?(%earl %pawn) (clan:title our))
+                      ?&  =(our (sein:title her))
+                          ::  XX is this necessary?
+                          ::
+                          !=(our her)
```

Comment:
That context is "if we're not a moon or a comet, we only include our deed if we're using asymmetric crypto with our sponsor". We only use asymmetric crypto until we've established a symmetric key. I could probably add a comment to that effect.

So that additional `!=(our her)` is just trying to avoid sending our deed to ourselves (which we only be possible if we were self-sponsored, ie a galaxy). Not particularly important either way, just needs to be consistent with `+deng`. I'll double check and comment accordingly.

As a side note, it's kind of weird that self communication goes all the way through the whole packet encryption mechanism. I guess it's the simplest way to make sure sequence numbers are maintained, message ordering within channels are preserved, etc.